shap_adapool/token_concatenation.py

```diff
@@ -29,3 +29,22 @@ def sentence_concat(array: NDArray[TokenDtype]) -> (NDArray[PhraseDtype], NDArra
                 token_buffer.write(token)
     index_map.append(-1)  # indicates the end of the array
     return sentences, index_map
+
+def kWord_concat(array: NDArray[TokenDtype], k: int) -> (NDArray[PhraseDtype], NDArray[IdxDtype]):
```

Comment:
Very nice job!!

One gripe I have here is the function naming. In Python the convention is to use `snake_case` naming for functions, so `kWord_concat` should be `k_word_concat`.

It is funny that some built-in modules like `unittest` use `camelCase` but that's only for historical reasons and (hopefully) bound to change at some point in the future.